qemu/tests/hotplug_mem_repeat.py

```diff
@@ -1,70 +1,58 @@
-import time
 import logging
 
 from virttest import error_context
 from virttest.utils_test.qemu import MemoryHotplugTest
 
 
 class MemoryHotplugRepeat(MemoryHotplugTest):
-
-    def start_test(self):
+    def repeat_hotplug(self, vm, target_mems):
         """
-        Prepare reqired test params, then start memory
-        hotplug/unplug tests in turn. And this is test entry.
+        Hotplug memory in target_mems
+        :param vm: vm object in this test
+        :param target_mems: target memory to be hotplugged
         """
-        target_mem = self.params["target_mem"]
-        vm = self.env.get_vm(self.params["main_vm"])
-        max_slots = int(self.params.get("slots_mem", 4))
-        scalability_test = self.params.get("scalability_test") == "yes"
-        repeats = scalability_test and max_slots or self.params["repeats"]
-        original_mem = self.get_guest_total_mem(vm)
-        for repeat in range(int(repeats)):
-            extra_params = (scalability_test and
-                            [{'slot_dimm': repeat}] or [None])[0]
-            error_context.context("Hotplug/unplug loop '%d'" % repeat,
-                                  logging.info)
-            self.turn(vm, target_mem, extra_params)
-            current_mem = self.get_guest_total_mem(vm)
-            if current_mem != original_mem:
-                self.test.fail("Guest memory changed about repeat"
-                               " hotpug/unplug memory %d times" % repeat)
-            time.sleep(1.5)
-        vm.verify_alive()
-        vm.reboot()
+        for target_mem in target_mems.split():
+            self.hotplug_memory(vm, target_mem)
+        self.check_memory(vm)
```

Comment:
Why don't check memory after each hotplug ?  and in this way, i am not very sure if it need wait a few seconds to make it take effect, need test.

Reply:
We could check memory every time we do hotplug, but it's not that meaningful.  The size for each memdev is only 128M(for x86),  which is a slight change and hardly noticeable due to the threshold.

Reply:
@huangyum  It is ok,  as i just realized `scalability_test` and `repeat_256` check memory at different timing (check after all mem be pluged  and check after every one be pluged), both of these scenario can be covered.
My concern is,  for `repeat_256`,  it do check memory after plug/unplug one memdev (128M), is it ok as you mentioned it is hardly noticeable due to the threshold ?   do me need to adjust threshold ?

Reply:
@huangyum  correct the comment above,  no `check_memory` for `repeat_256`.  it use  `hotplug_memory`, `unplug_memory` directly. ignore it.